superset-frontend/src/explore/actions/saveModalActions.ts

```diff
@@ -0,0 +1,321 @@
+            sliceName,
+          ),
+        ),
+      );
+    } else {
+      toasts.push(
+        addSuccessToast(
+          t(
+            'Chart [%s] was added to dashboard [%s]',
+            sliceName,
+            addedToDashboard.title,
+          ),
+        ),
+      );
+    }
+  }
+
+  return toasts;
+};
+
+export const updateSlice =
+  (
+    slice: Slice,
+    sliceName: string,
+    dashboards: number[],
+    addedToDashboard?: {
+      title: string;
+      new?: boolean;
+    },
+  ) =>
+  async (dispatch: Dispatch, getState: () => Partial<QueryFormData>) => {
+    const { slice_id: sliceId, owners, form_data: formDataFromSlice } = slice;
+    const formData = getState().explore?.form_data;
+    try {
+      const response = await SupersetClient.put({
+        endpoint: `/api/v1/chart/${sliceId}`,
+        jsonPayload: getSlicePayload(
+          sliceName,
+          formData,
+          dashboards,
+          owners as [],
+          formDataFromSlice,
+        ),
+      });
+
+      dispatch(saveSliceSuccess(response.json));
+      addToasts(false, sliceName, addedToDashboard).map(dispatch);
+      return response.json;
+    } catch (error) {
+      dispatch(saveSliceFailed());
+      throw error;
+    }
+  };
+
+export const createSlice =
+  (
+    sliceName: string,
+    dashboards: number[],
+    addedToDashboard?: {
+      title: string;
+      new?: boolean;
+    },
+  ) =>
+  async (dispatch: Dispatch, getState: () => Partial<QueryFormData>) => {
+    const formData = getState().explore?.form_data;
+    try {
+      const response = await SupersetClient.post({
+        endpoint: `/api/v1/chart/`,
+        jsonPayload: getSlicePayload(
+          sliceName,
+          formData,
+          dashboards,
+          [],
+          {} as QueryFormData,
+        ),
+      });
+
+      dispatch(saveSliceSuccess(response.json));
+      addToasts(true, sliceName, addedToDashboard).map(dispatch);
+      return response.json;
+    } catch (error) {
+      dispatch(saveSliceFailed());
+      throw error;
+    }
+  };
+
+export const createDashboard =
+  (dashboardName: string) => async (dispatch: Dispatch) => {
+    try {
+      const response = await SupersetClient.post({
+        endpoint: `/api/v1/dashboard/`,
+        jsonPayload: { dashboard_title: dashboardName },
+      });
+
+      return response.json;
+    } catch (error) {
+      dispatch(saveSliceFailed());
+      throw error;
+    }
+  };
+
+export const getSliceDashboards =
+  (slice: Partial<Slice>) => async (dispatch: Dispatch) => {
+    try {
+      const response = await SupersetClient.get({
+        endpoint: `/api/v1/chart/${slice.slice_id}?q=${rison.encode({
+          columns: ['dashboards.id'],
+        })}`,
+      });
+
+      return response.json.result.dashboards.map(
+        ({ id }: { id: number }) => id,
+      );
+    } catch (error) {
+      dispatch(saveSliceFailed());
+      throw error;
+    }
+  };
+export { QueryFormData };
```

Comment:
@EnxDev this is creating some warning. Do we need it?

Reply:
I removed it here https://github.com/apache/superset/pull/28786